Limit fusion to ops with single uses to avoid duplication

ThomasRaoux opened the pull request.

ThomasRaoux commented:
Now that the noise problem are fixed in the benchmark, limiting the fusion to not duplicate ops seems to not cause regressions. So sending it for review.

MaheshRavishankar commented on iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp:
```diff
@@ -1127,6 +1127,11 @@ static unsigned decideFusableLinalgOps(mlir::FuncOp funcOp) {
                 consumerIndexingMap.getResults()) {
           continue;
         }
+        if (llvm::any_of(
```
WHy this?

MaheshRavishankar replied:
I have to see some examples here. These heuristics have become really long and patch-worky. Need to resolve this and do some house cleaning here.

Ill stamp this for now.

antiagainst commented:
FYI: I fixed a bug that caused benchmark commenting not working with https://github.com/google/iree/pull/7112. So you may want to merge the latest `main` branch to give it a go. Sorry about the inconvenience!

ThomasRaoux commented:
> FYI: I fixed a bug that caused benchmark commenting not working with #7112. So you may want to merge the latest `main` branch to give it a go. Sorry about the inconvenience!

Thanks, I rebased

ThomasRaoux commented:
I fixed the performance issue on MobileNetV3, the problem was that broadcast ops were not getting fused anymore so I changed the code to make sure broadcast are fused even if they have multiple uses. Looking at the remaining regressions it looks like noise to me. Looks like the last result from `MobileSSD` is an outlier and the perf I get matches the rest of the measures. The DeepLabV3 perf on mali is the only one I'm unsure about, I'll check if it is a real regression.

ThomasRaoux commented:
Compiling DeepLabV3 model locally my patch doesn't change the output at all so it cannot change performance. I think it should be fine to push this. @antiagainst, @MaheshRavishankar let me know if you have any concerns.

antiagainst commented:
SGTM. Thanks for double checking it, Thomas!
